app/resources/sli/sources/base.py

```diff
@@ -0,0 +1,223 @@
+import dataclasses
+import datetime
+import enum
+import inspect
+from decimal import Decimal
+from typing import Dict, List, Optional, Tuple, Union
+
+
+class SourceError(Exception):
+    pass
+
+
+class TimeRange:
+    DEFAULT: "TimeRange"
```

Comment:
setting it to string might erroneously suggest to others that this variable will be a string, whereas it's a class instance. What do you think about setting it to NotImplementedError or something like that?

Reply:
oh oops I read it as `DEFAULT=`